app/controllers/three_d_secure_controller.js

```diff
@@ -98,16 +98,26 @@ module.exports = {
     const md = _.get(charge, 'auth3dsData.md')
     const htmlOut = _.get(charge, 'auth3dsData.htmlOut')
     const worldpayChallengeJwt = _.get(charge, 'auth3dsData.worldpayChallengeJwt')
+    const threeDSReturnUrl = `${req.protocol}://${req.hostname}${paths.generateRoute('external.card.auth3dsRequiredIn', { chargeId: charge.id })}`
 
     if (issuerUrl && paRequest) {
       const data = {
         postUrl: issuerUrl,
         paRequest: paRequest,
-        threeDSReturnUrl: `${req.protocol}://${req.hostname}${paths.generateRoute('external.card.auth3dsRequiredIn', { chargeId: charge.id })}`
+        threeDSReturnUrl: threeDSReturnUrl
       }
       if (md) {
         data.md = md
       }
+
+      logger.info('Form details sent to card issuer', {
```

Comment:
The form hasn’t actually been submitted yet (it probably will be but something could go wrong) and it would probably be useful to specifically name the form we’re loading, so maybe this could be something like: “Rendering form to posts to card issuer or gateway to initiate 3DS authentication”.

Reply:
Done